galley/pkg/server/server.go

```diff
@@ -229,6 +238,13 @@ func (s *Server) Run() {
 			scope.Fatalf("Galley Server unexpectedly terminated: %v", err)
 		}
 	}()
+	if s.callOut != nil {
+		s.serveWG.Add(1)
+		go func() {
+			defer s.serveWG.Done()
+			s.callOut.Run()
+		}()
+	}
```

Comment:
With your changes s.serveWG.Wait() probably needs to be if s.grpcServer != nil && s.callout!= nil in Close() method.